Trading/Exchange/binance/binance_exchange.py

```diff
@@ -37,16 +72,166 @@ def get_name(cls):
+        # only enable the trading type with the majority of asked symbols
+        # todo remove this and use both types when exchange-side multi portfolio is enabled
+        linear_count = symbol_counts.get(trading_enums.FutureContractType.LINEAR_PERPETUAL.value, 0)
+        inverse_count = symbol_counts.get(trading_enums.FutureContractType.INVERSE_PERPETUAL.value, 0)
+        if linear_count >= inverse_count:
+            account_types.append(self.LINEAR_TYPE)   # allows to fetch linear markets
+            if inverse_count:
+                exchange_manager.logger.error(
+                    f"For now, due to the inverse and linear portfolio split on Binance Futures, OctoBot only "
+                    f"supports either linear or inverse trading at a time. Ignoring {inverse_count} inverse "
+                    f"futures trading pair as {linear_count} linear futures trading pairs are enabled."
+                )
+        else:
+            account_types.append(self.INVERSE_TYPE)  # allows to fetch inverse markets
+            if linear_count:
+                exchange_manager.logger.error(
+                    f"For now, due to the inverse and linear portfolio split on Binance Futures, OctoBot only "
+                    f"supports either linear or inverse trading at a time. Ignoring {linear_count} linear "
+                    f"futures trading pair as {inverse_count} inverse futures trading pairs are enabled."
+                )
+        return account_types
+
+    @classmethod
+    def get_supported_exchange_types(cls) -> list:
+        """
+        :return: The list of supported exchange types
+        """
+        return [
+            trading_enums.ExchangeTypes.SPOT,
+            trading_enums.ExchangeTypes.FUTURE,
+        ]
+
     def get_additional_connector_config(self):
-        return {
+        config = {
             ccxt_constants.CCXT_OPTIONS: {
                 "quoteOrderQty": False,  # disable quote conversion
                 "recvWindow": 60000,    # default is 10000, avoid time related issues
             }
         }
+        if self.exchange_manager.is_future:
+            config[ccxt_constants.CCXT_OPTIONS]['fetchMarkets'] = self._futures_account_types
+        return config
+
+    @classmethod
+    def update_supported_elements(cls, exchange_manager):
+        if exchange_manager.is_future:
+            cls.UNSUPPORTED_ORDERS = cls.FUTURES_UNSUPPORTED_ORDERS
+        else:
+            cls.UNSUPPORTED_ORDERS = cls.SPOT_UNSUPPORTED_ORDERS
+
+    async def get_balance(self, **kwargs: dict):
+        if self.exchange_manager.is_future:
+            balance = []
+            for account_type in self._futures_account_types:
+                balance.append(await self.connector.get_balance(**kwargs, subType=account_type))
+            # todo remove this and use both types when exchange-side multi portfolio is enabled
+            return balance[0]   # only returning linear portfolio
```

Comment:
Only linear?

Reply:
this comment is wrong, i'll fix it. the idea is to return the 1st fetched (there will only be 1 as long as both linear and inverse are not supported simultaneously)